src/api.ts

```diff
@@ -3855,6 +3856,159 @@ export const methods = {
+        callback(null, response)
+        countSuccessResponse(api_name, 'success', 'archiver')
+      } else {
+        logEventEmitter.emit('fn_end', ticket, { success: false }, performance.now())
+        callback({ code: -32000, message: 'Network account data not found or invalid' }, null)
+        countFailedResponse(api_name, 'Network account data not found or invalid')
+      }
+    } catch (error: any) {
+      console.error('Error fetching network account data:', error)
+      logEventEmitter.emit('fn_end', ticket, { success: false }, performance.now())
+      callback({ code: -32000, message: 'Failed to fetch network account data', data: error.message }, null)
+      countFailedResponse(api_name, 'Failed to fetch network account data')
+    }
+  },
+  shardeum_getCycleInfo: async function (args: RequestParamsLike, callback: JSONRPCCallbackTypePlain) {
+    const api_name = 'shardeum_getCycleInfo'
+    nestedCountersInstance.countEvent('endpoint', api_name)
+
+    if (
+      !Array.isArray(args) ||
+      args.length > 1 ||
+      (args.length === 1 && typeof args[0] !== 'number' && args[0] !== null)
+    ) {
+      callback({ code: -32602, message: 'Invalid params: Expected a single number or null.' }, null)
+      countFailedResponse(api_name, 'Invalid params: Expected a single number or null.')
+      return
+    }
+    const ticket = crypto
+      .createHash('sha1')
+      .update(api_name + Math.random() + Date.now())
+      .digest('hex')
+
+    logEventEmitter.emit('fn_start', ticket, api_name, performance.now())
+    const cycleNumber = args.length === 1 ? args[0] : null
+    try {
+      const result = await collectorAPI.getCycleInfo(cycleNumber)
+
+      if (result) {
+        const cycleRecord = result.cycleRecord
+        // transform the response to match the spec structure
+        const restructuredData = {
+          cycleCounter: cycleRecord.counter,
+          startTime: cycleRecord.start * 1000,
+          endTime: (cycleRecord.start + cycleRecord.duration) * 1000,
+          nodes: {
+            active: cycleRecord.active,
+            standby: cycleRecord.standby,
+            syncing: cycleRecord.syncing,
+          },
+          desired: cycleRecord.desired,
+          duration: cycleRecord.duration,
+          maxSyncTime: cycleRecord.maxSyncTime,
+          timestamp: cycleRecord.timestamp,
+        }
+
+        const response = {
+          cycleInfo: restructuredData,
+        }
+
+        logEventEmitter.emit('fn_end', ticket, { success: true }, performance.now())
+        callback(null, response)
+        countSuccessResponse(api_name, 'success', 'collector')
+      } else {
+        logEventEmitter.emit('fn_end', ticket, { success: false }, performance.now())
+        callback({ code: -32000, message: 'Cycle info not found' }, null)
+        countFailedResponse(api_name, 'Cycle info not found')
+      }
+    } catch (error: any) {
```

Comment:
The `else` block and `catch` block are redundant. Both of them log the error as cycle info not found. The catch block should be used for handling unforeseen or other wider errors